addons/preview-project-description/userscript.js

```diff
@@ -148,77 +128,22 @@ export default async function ({ addon, console, msg }) {
-      // is exported as module.exports.View: https://github.com/scratchfoundation/scratch-www/blob/59fa9cd12744ec6fcfc73d0d92c4fbbe5ca73d38/src/views/preview/project-view.jsx#L1213
-      try {
-        const stack = new Error().stack;
-        // In Firefox, `View` is part of the call stack returned with new Error().stack, so we just check if it's there.
-        // In Chrome, `View` is not listed, so instead we look for `Function.mapToProps`, which is internally used by Redux's
-        // connect() implementation: https://github.com/reduxjs/react-redux/blob/v5.0.7/src/connect/wrapMapToProps.js#L43
-        return Boolean(
-          stack
-            .split("\n")
-            .slice(0, 4) // No need to look deeper than 4 lines of the error stack.
-            .find((s) => s.trimStart().startsWith("at Function.mapToProps ") || s.startsWith("e.exports.View"))
-        );
-      } catch {
-        // Not sure if anything could theoretically throw inside the "try", but let's be careful
-        return false;
-      }
-    };
-
-    let step = 0;
-    // https://github.com/scratchfoundation/scratch-www/blob/59fa9cd12744ec6fcfc73d0d92c4fbbe5ca73d38/src/views/preview/project-view.jsx#L1028
-    // 1. A call to author.id.toString() with a matching function stack.
-    // 2. A call to state.session.user.id.toString() with a matching function stack (author ID = user ID).
-    // 3. A getter to `state.session.session.user.username` on the same event loop cycle. Reset to step 0.
-
-    const originalNumberToString = Number.prototype.toString;
-    Number.prototype.toString = function (...args) {
-      if (!currentlyRerendering) return originalNumberToString.apply(this, args);
-      if (this !== redux.state?.session?.session?.user?.id) return originalNumberToString.apply(this, args);
-      if (!beingCalledByMapStateToProps()) return originalNumberToString.apply(this, args);
-
-      step++;
-      if (step === 3) {
-        // Huh, weird
-        step = 0;
-        return originalNumberToString.apply(this, args);
-      }
-
-      queueMicrotask(() => (step = 0));
-
-      return originalNumberToString.apply(this, args);
+    // Override the render function of the Preview component
+    // https://github.com/scratchfoundation/scratch-www/blob/fdcb700/src/views/preview/project-view.jsx
+    const reactRootElement = document.querySelector("#app");
+    const reactContainerKey = Object.keys(reactRootElement).find((key) => key.startsWith(REACT_CONTAINER_PREFIX));
+    let instance = reactRootElement[reactContainerKey];
+    while (!instance.stateNode || typeof instance.stateNode.handleUpdateProjectId !== "function") {
+      instance = instance.child;
+    }
+    const PreviewComponent = instance.stateNode.constructor;
+    const oldRender = PreviewComponent.prototype.render;
+    PreviewComponent.prototype.render = function () {
+      const oldProps = this.props;
+      this.props = { ...this.props, isEditable: !currentlyEnabled };
```

Comment:
Excellent! Thanks mxmou

Messages from 3 months ago lol:

![image](https://github.com/ScratchAddons/ScratchAddons/assets/17484114/2b817990-3896-43a5-8313-6516f1796f81)

Reply:
(the folders addon did use this pattern apparently, but I missed it!)